test/src/dgkernels/DGCoupledDiffusion.C

```diff
@@ -0,0 +1,133 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "DGCoupledDiffusion.h"
+
+// MOOSE includes
+#include "MooseVariable.h"
+#include "SystemBase.h"
+
+// libMesh includes
+#include "libmesh/utility.h"
+
+template <>
+InputParameters
+validParams<DGCoupledDiffusion>()
+{
+  InputParameters params = validParams<DGKernel>();
+  // See header file for sigma and epsilon
+  params.addRequiredParam<Real>("sigma", "sigma");
+  params.addRequiredParam<Real>("epsilon", "epsilon");
+  params.addParam<MaterialPropertyName>(
+      "diff", 1., "The diffusion (or thermal conductivity or viscosity) coefficient.");
+  params.addRequiredCoupledVar("v", "The governing variable that controls diffusion of u.");
+  return params;
+}
+
+DGCoupledDiffusion::DGCoupledDiffusion(const InputParameters & parameters)
+  : DGKernel(parameters),
+    _epsilon(getParam<Real>("epsilon")),
+    _sigma(getParam<Real>("sigma")),
+    _diff(getMaterialProperty<Real>("diff")),
+    _diff_neighbor(getNeighborMaterialProperty<Real>("diff")),
+    _v_var(*getVar("v", 0)),
+    _v(_v_var.sln()),
+    _v_neighbor(_v_var.slnNeighbor()),
+    _grad_v(_v_var.gradSln()),
+    _grad_v_neighbor(_v_var.gradSlnNeighbor()),
+    _v_id(coupled("v"))
+{
+}
+
+Real
+DGCoupledDiffusion::computeQpResidual(Moose::DGResidualType type)
+{
+  Real r = 0;
+
+  const unsigned int elem_b_order = _v_var.order();
+  const double h_elem =
```

Comment:
We always use `Real` in MOOSE code. `double` should only appear in utility code that doesn't include any part of MOOSE.